### ISSUE TITLE
How to pass props in m-for?

### ISSUE THREAD

**Issue by FNGR2911** [kbrsh/moon#201]
Hi everybody! I can't figure out how to pass props into a component inside a m-for directive!? this is how my code looks:

```javascript
  Moon.component('storeitem', {
    props: ['store'],
    template: `<li class="storelist__item">
        <strong>{{ store.name }}</strong>,<br>
        {{ store.street }} {{ store.active }}
      </li>`,
  })

  const app = new Moon({
    el: '#app',
    template: `<ul class="storelist">
      <storeitem m-for="store, index in stores" store="{{store}}"></storeitem>
    </ul>`,
    data: {
      stores,
    },
  })
```
I hope you can see what i mean. thank you!

**Comment by kbrsh** [comment]
Hey! Looks like you are passing the `store` prop correctly—but you are passing it as a _string_. The `m-literal` directive can be used for passing other types.

```html
<ul class="storelist">
  <storeitem m-for="store, index in stores" m-literal:store="store"></storeitem>
</ul>
```

**Comment by FNGR2911** [comment]
thank you @kbrsh! That works perfect. Haven't seen the m-literal. Sorry for asking, but now I got another problem I can't figure out. When I try to emit an event via my compoment I get an error:

```javascript
Uncaught TypeError: handlers[i] is not a function
```

This is how my code looks:
```javascript
  Moon.component('storeitem', {
    props: ['store'],
    methods: {
      clickHandler: function() {
        console.log('called component method')
        this.emit('storeClicked')
      },
    },
    template: `<li class="storelist__item" m-on:click="clickHandler">
        <strong>{{ store.name }}</strong>,<br>
        {{ store.street }}, {{ store.zip }} {{ store.city }}<br>
        {{ store.active }}
      </li>`,
  })

  const app = new Moon({
    el: '#app',
    template: `<ul class="storelist">
        <storeitem m-for="store, index in stores" m-literal:store="store" m-on:storeClicked="setActiveStore(store)"></storeitem>
      </ul>`,
    data: {
      stores,
    },
    methods: {
      setActiveStore: function(item) {
        console.log('called app method')
        const stores = this.get('stores').map(store => {
          store.id === item.id ? (store.active = true) : (store.active = false)
          return store
        })
        this.set('stores', stores)
      },
    },
  })
```
Can't see what I'm doing wrong. Looks like your demo code in the docs!?

thank you very much!

**Comment by kbrsh** [comment]
No problem! @FNGR2911

That's a bug in v0.11.0 that will be fixed in v1.0.0 once I do a rewrite of the way Moon works. The current way is fundamentally flawed in many ways and this is one of them.

I'm closing this issue, as the original problem has been resolved and the current one has another issue open. Feel free to ask any more questions in the [Moon Slack](https://moon-slack.herokuapp.com).
